File: leetcode/1001-2000/1301-1400/1137-n-th-tribonacci-number/NThTribonacciNumber.java
public class NThTribonacciNumber {
  public static void main(String[] args) {
    System.out.println(new Solution().tribonacci(25));
  }
}

class Solution {
  public int tribonacci(int n) {
    int t0 = 0;
    int t1 = 1;
    int t2 = 1;
    int t = 2;

    while (n-- > 2) {
      t = t0 + t1 + t2;

      t0 = t1;
      t1 = t2;
      t2 = t;
    }

    return t;
  }
}

File: leetcode/1001-2000/1301-1400/1137-n-th-tribonacci-number/NThTribonacciNumber.ts
function tribonacci(n: number): number {
  if (n === 0) {
    return 0;
  }

  if (n === 1) {
    return 1;
  }

  if (n === 2) {
    return 1;
  }

  let t0 = 0;
  let t1 = 1;
  let t2 = 1;

  let t = 0;
  while (n-- > 2) {
    t = t0 + t1 + t2;
    t0 = t1;
    t1 = t2;
    t2 = t
  }

  return t2;
};

console.log(tribonacci(25));
